File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.java
class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = new int[2];
        int cnt_0 = 0;
        int cnt_a = 0;
        
        for(int i : lottos){
            if(i == 0){
                cnt_0++;
            }else{
                for(int j : win_nums){
                    if(j == i){
                        cnt_a++;
                        break;
                    }
                }
            }
        }
        int min = cnt_a;
        int max = cnt_a + cnt_0;
        answer[0] = Math.min(7-max, 6);
        answer[1] = Math.min(7-min, 6);
        return answer;
    }
}

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    low = 0
    high = 0
    for i in lottos:
        if i in win_nums:
            low += 1
            high += 1
        elif i == 0:
            high += 1
    low = 7-low if 7-low < 6 else 6
    high = 7-high if 7-high < 6 else 6
    answer = [high, low]
    return answer